tests/test_hierarchy.cc

```diff
@@ -47,6 +47,107 @@ double Source<dim>::value(dealii::Point<dim> const &, unsigned int const) const
   return 0.;
 }
 
+template <int dim>
+class ConstantMaterialProperty : public dealii::Function<dim>
+{
+public:
+  ConstantMaterialProperty() = default;
+
+  virtual double value(dealii::Point<dim> const &p,
+                       unsigned int const component = 0) const override final;
+};
+
+template <int dim>
+double ConstantMaterialProperty<dim>::value(dealii::Point<dim> const &,
+                                            unsigned int const) const
+{
+  return 1.;
+}
+
+template <int dim>
+class LinearXMaterialProperty : public dealii::Function<dim>
+{
+public:
+  LinearXMaterialProperty() = default;
+
+  virtual double value(dealii::Point<dim> const &p,
+                       unsigned int const component = 0) const override final;
+};
+
+template <int dim>
+double LinearXMaterialProperty<dim>::value(dealii::Point<dim> const &p,
+                                           unsigned int const) const
+{
+  return 1. + p[0];
+}
+
+template <int dim>
+class LinearMaterialProperty : public dealii::Function<dim>
+{
+public:
+  LinearMaterialProperty() = default;
+
+  virtual double value(dealii::Point<dim> const &p,
+                       unsigned int const component = 0) const override final;
+};
+
+template <int dim>
+double LinearMaterialProperty<dim>::value(dealii::Point<dim> const &p,
+                                          unsigned int const) const
+{
+  double value = 1.;
+  for (unsigned int d = 0; d < dim; ++d)
+    value += p[d];
+
+  return value;
+}
+
+template <int dim>
+class DiscontinuousMaterialProperty : public dealii::Function<dim>
+{
+public:
+  DiscontinuousMaterialProperty() = default;
+
+  virtual double value(dealii::Point<dim> const &p,
+                       unsigned int const component = 0) const override final;
+};
+
+template <int dim>
+double DiscontinuousMaterialProperty<dim>::value(dealii::Point<dim> const &p,
+                                                 unsigned int const) const
+{
+  double value = 10.;
+  for (unsigned int d = 0; d < dim; ++d)
+    if (p[d] > 0.5)
+      value *= value;
```

Comment:
A more challenging problem would be chess coloring. I also remember some multigrid methods (blackbox?) were tested on some problem where they made the discontinuity not grid aligned. Will need to find it out (I think it may be in the Proceedings of DD20).